js/plugins/google-cloud/src/gcpOpenTelemetry.ts

```diff
@@ -307,6 +307,26 @@ class AdjustingTraceExporter implements SpanExporter {
       )
     );
 
+    // Check if we have a failure in the root span that requires special handling
```

Comment:
Should we throw this into a helper function like the ones below? Either that, we should remove the span = this.markFailedSpan call below.

Reply:
That helper and all the others run over all the individual child spans, so it's still relevant.